packages/flutter/lib/src/material/carousel.dart

```diff
@@ -0,0 +1,762 @@
+    }
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final AxisDirection axisDirection = _getDirection(context);
+    final ScrollPhysics physics = widget.itemSnapping
+      ? const CarouselScrollPhysics()
+      : ScrollConfiguration.of(context).getScrollPhysics(context);
+    final EdgeInsets effectivePadding = widget.padding ?? const EdgeInsets.all(4.0);
+    final Color effectiveBackgroundColor = widget.backgroundColor ?? Theme.of(context).colorScheme.surface;
+    final double effectiveElevation = widget.elevation ?? 0.0;
+    final ShapeBorder effectiveShape = widget.shape
+      ?? const RoundedRectangleBorder(
+        borderRadius: BorderRadius.all(Radius.circular(28.0))
+      );
+
+    final List<Widget> children = List<Widget>.generate(widget.children.length, (int index) {
+      return Padding(
+        padding: effectivePadding,
+        child: Material(
+          clipBehavior: Clip.antiAlias,
+          color: effectiveBackgroundColor,
+          elevation: effectiveElevation,
+          shape: effectiveShape,
+          child: Stack(
+            fit: StackFit.expand,
+            children: <Widget>[
+              widget.children.elementAt(index),
+              Material(
+                color: Colors.transparent,
+                child: InkWell(
+                  onTap: () {
+                    widget.onTap?.call(index);
+                  },
+                  overlayColor: widget.overlayColor ?? WidgetStateProperty.resolveWith((Set<WidgetState> states) {
+                    if (states.contains(WidgetState.pressed)) {
+                      return theme.colorScheme.onSurface.withOpacity(0.1);
+                    }
+                    if (states.contains(WidgetState.hovered)) {
+                      return theme.colorScheme.onSurface.withOpacity(0.08);
+                    }
+                    if (states.contains(WidgetState.focused)) {
+                      return theme.colorScheme.onSurface.withOpacity(0.1);
+                    }
+                    return null;
+                  }),
+                ),
+              ),
+            ],
+          ),
+        ),
+      );
+    });
+
+    return Scrollable(
+      axisDirection: axisDirection,
+      controller: _controller,
+      physics: physics,
+      viewportBuilder: (BuildContext context, ViewportOffset position) {
+        return Viewport(
+          cacheExtent: 0.0,
+          cacheExtentStyle: CacheExtentStyle.viewport,
+          axisDirection: axisDirection,
+          offset: position,
+          clipBehavior: Clip.antiAlias,
+          slivers: <Widget>[
+            _SliverFixedExtentCarousel(
+              itemExtent: itemExtent!,
+              minExtent: widget.shrinkExtent ?? 0.0,
+              delegate: SliverChildBuilderDelegate(
+                (BuildContext context, int index) {
+                  return children.elementAt(index);
```

Comment:
Could we create the children on-demand right here instead of pre-building the entire `children` list?

Reply:
Updated!